docs/root/configuration/listeners/network_filters/connection_limit_filter.rst

```diff
@@ -0,0 +1,106 @@
+Each connection processed by the filter marked as an active connection, and if the number of active connections reaches the max connections limit,
+the connection will be closed without further filter iteration.
+
+-  The connection limit filter is similar to the L4 local rate limit filter, but instead of enforcing the limit on connections rate, the filter limits the number of active connections.
+-  The filter maintains an atomic counter of active connection count. It has a max connections limit value based on the configured total number of connections.
+   When a new connection request comes, the filter tries to increment the connection counter. The connection is allowed if the counter is less than the max connections limit, otherwise the connection gets rejected.
+   When an active connection is closed, the filter decrements the active connection counter.
+-  The filter does not stop connection creation but will close the connections that were accepted but were deemed as overlimit.
+-  **Slow rejection:** The filter can stop reading from the connection and close it after a delay instead of rejecting it right away or letting requests go through before the rejection.
+   This way we can prevent a malicious entity from opening new connections while draining their resources.
+
+.. note::
+  In the current implementation each filter chain has an independent connection limit.
+
+Algorithm
+---------
+
+The filter will use the reference counting algorithm to keep trace of active connection count.
+
+1. Active connection count < max connection limit:
+
+   -  Increment count for a new allowed connection.
+   -  Decrement count when a connection closes.
+
+2. Active connection count >= max connection limit:
+
+   -  Close the new connection request after configured delay time.
+
+API Reference
+-------------
+
+[extensions.filters.network.connection_limit.v3.ConnectionLimit proto]
+
+{
+
+**“stat_prefix”**: “…”,
+
+**“max_connections”**: “…”,
+
+**“delay”**: “{…}”,
+
+**“runtime_enabled”**: “{…}”
+
+}
+
+**stat_prefix**
+
+(`string <https://developers.google.com/protocol-buffers/docs/proto#scalar>`_)
+The prefix to use when emitting statistics.
+
+**max_connections**
+
+(`uint64 <https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#google.protobuf.UInt64Value>`_)
+The limit supplied in max connections.
+
+**delay**
+
+(`Duration <https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#duration>`_)
+The delay in milliseconds for the slow rejection connections. If not set, this defaults to 0ms.
+
+**runtime_enabled**
+
+(`config.core.v3.RuntimeFeatureFlag <https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/base.proto#envoy-v3-api-msg-config-core-v3-runtimefeatureflag>`_)
+Runtime flag that controls whether the filter is enabled or not. If not specified, defaults to enabled.
```

Comment:
This can all be deleted as it's duplicated in the proto docs. Just reference the proto docs above or the fields directly as we do elsewhere.

Reply:
Deleted.